Add option for red to green color scale

SammyRamone opened the pull request:
I understand that the current color scale should show a heatmap. However, everytime I showed an image to one of my coworkers, they were confused by the colors. They always expected areas with low score as red and areas with good score as green. Especially in combination with the RViz visualization.
Therefore, I implemented an option to switch from the heatmap color scheme to a red to green color scaling.
I don't know, if you want to have this in the main code base. If not, then that is also totally fine for me and you can close this PR.
If you want to merge it, I can also add some documentation to the PR.

Previous color scheme (with use_full_color_range)
![standard](https://github.com/ros-industrial/reach/assets/11520148/93a023c7-5e04-4842-8e52-73924ec1565d)

New red to green option (with use_full_color_range)
![red_to_green](https://github.com/ros-industrial/reach/assets/11520148/dfce54ee-7663-46cb-8940-9721114a019e)

marip8 commented:
I am fairly red-green colorblind so it's not terribly helpful for me personally, but I don't oppose the ability to change the color scheme. Instead of adding a boolean to toggle between blue->red and red->green, I would prefer to add arguments to parameterize `h_min` and `h_max` and give them default values of 0.0 and 0.75*360=270, respectively. Then users can change `h_min` and `h_max` to be whatever they want. We could consider adding some logic to to handle the case where `h_max` < `h_min` (e.g. red -> blue) too

SammyRamone commented:
Yeah it makes sense to do it more general.
Maybe let the user define a `hue_low_score` and a `hue_high_score`. The direction and the `h_min` & `h_max` can then be set based on this.
I will change the PR as soon as I have time for it.

SammyRamone commented:
I changed the PR to accept `hue_low_score` and `hue_high_score`. The default behavior is still the same as before (heatmap from blue->green->red). I also added a bit of documentation in reach_ros2 package.
It is now also possible to create color scales without any green. For example this red->blue one with `hue_low_score: 360.0` and `hue_high_score: 220.0`, see below.
![grafik](https://github.com/ros-industrial/reach/assets/11520148/c2b45289-6532-4c30-bb77-1f6037f60cd0)

Handling of `hue_high_score < hue_low_score` is no problem. However, currently the hue circle is never traversed from 360° to 0° and vice versa. If you want this feature too, I can add it, but as it would require yet another parameter and makes things more confusing for the user, I omitted it for now.

marip8 commented on include/reach/types.h:
```diff
@@ -124,7 +124,7 @@ std::vector<float> normalizeScores(const ReachResult& result, bool use_full_rang
  * @return An array of RGB colors on [0, 1] for each channel
  */
-Eigen::MatrixX3f computeHeatMapColors(const std::vector<float>& scores);
+Eigen::MatrixX3f computeHeatMapColors(const std::vector<float>& scores, float hue_low_score, float hue_high_score);
```
```suggestion
Eigen::MatrixX3f computeHeatMapColors(const std::vector<float>& scores, float hue_low_score = 270.0f, float hue_high_score = 0.0f);
```

marip8 commented on include/reach/types.h:
```diff
@@ -134,7 +134,8 @@ Eigen::MatrixX3f computeHeatMapColors(const std::vector<float>& scores);
  * @return An array of RGB colors on [0, 1] for each channel
  */
-Eigen::MatrixX3f computeHeatMapColors(const ReachResult& result, bool use_full_color_range);
+Eigen::MatrixX3f computeHeatMapColors(const ReachResult& result, bool use_full_color_range, float hue_low_score,
+                                      float hue_high_score);
```
```suggestion
Eigen::MatrixX3f computeHeatMapColors(const ReachResult& result, bool use_full_color_range, float hue_low_score = 270.0f,
                                      float hue_high_score = 0.0f);
```

marip8 commented on src/python/python_bindings.cpp:
```diff
@@ -128,19 +128,21 @@ bp::list normalizeScoresPython(const ReachResult& result, bool use_full_range)
 }
 
-np::ndarray computeHeatMapColorsPython1(const ReachResult& result, bool use_full_color_range)
+np::ndarray computeHeatMapColorsPython1(const ReachResult& result, bool use_full_color_range, float hue_low_score,
+                                        float hue_high_score)
```
```suggestion
np::ndarray computeHeatMapColorsPython1(const ReachResult& result, bool use_full_color_range, float hue_low_score = 270.0f,
                                        float hue_high_score = 0.0f)
```

marip8 commented on src/python/python_bindings.cpp:
```diff
@@ -128,19 +128,21 @@ bp::list normalizeScoresPython(const ReachResult& result, bool use_full_range)
 }
 
-np::ndarray computeHeatMapColorsPython2(const bp::list& scores)
+np::ndarray computeHeatMapColorsPython2(const bp::list& scores, float hue_low_score, float hue_high_score)
```
```suggestion
np::ndarray computeHeatMapColorsPython2(const bp::list& scores, float hue_low_score = 270.0f, float hue_high_score = 0.0f)
```

marip8 reviewed (commented):
I think this looks good. Let's just add default values for the min and max hues where we changed the function definitions so we can maintain API compatibility.

marip8 reviewed (approved):
Looks good. Thanks for the contribution
